pixeltable/utils/sample.py

```diff
@@ -1,88 +1,98 @@
 from __future__ import annotations
 
+import dataclasses
 from typing import Optional
 
 import sqlalchemy as sql
 
-from pixeltable import exprs, type_system as ts
+from pixeltable import type_system as ts
 from pixeltable.exprs.expr import DataRow, Expr
 from pixeltable.exprs.literal import Literal
 from pixeltable.exprs.row_builder import RowBuilder
 from pixeltable.exprs.sql_element_cache import SqlElementCache
 
 
-class SampleClause(Expr):
-    """
-    An `Expr` that computes a sample of the associated table.
-    """
+@dataclasses.dataclass
+class SampleClause:
+    """Defines a sampling clause for a table."""
 
-    CURRENT_VERSION = 1
+    version: Optional[int]
+    n: Optional[int]
+    n_per_stratum: Optional[int]
+    fract: Optional[float]
+    seed: Optional[int]
+    stratify_list: Optional[list[Expr]]
```

Comment:
Elsewhere in the codebase, we generally name list[Expr] as '<something>_exprs'. It would be good to follow that convention.

Reply:
Except `exprs_` not `_exprs`. (Anything *preceded* by an underscore will be treated by the linter as an anonymous variable.)

But personally I prefer `stratify_list` here. The generic `exprs_` is fine when it's a list of expressions with no specific meaning (such as in `InlineList`), but here I think it makes the code clearer if the variable is named for what it is. I'd vote for keeping as is.